aten/src/ATen/native/quantized/cpu/qnnpack/src/deconv-run.cc

```diff
@@ -1,11 +1,75 @@
 
+#include <cstring>
+#include <memory>
+
+#include <conv_utils.h>
+#include <pytorch_qnnpack.h>
 #include <qnnpack/indirection.h>
-#include <qnnpack/pack.h>
-#include <qnnpack_func.h>
+#include <qnnpack/log.h>
+#include <qnnpack/math.h>
+#include <qnnpack/params.h>
 
 namespace qnnpack {
+struct q8conv_context {
+  size_t bs;
+  size_t ks;
+  size_t kc;
+  size_t kc_stride;
+  size_t m;
+  size_t m_stride;
+  size_t n;
+  size_t n_stride;
+  const uint8_t** indirect_a;
+  const void* packed_w;
+  uint8_t* c;
+  size_t c_stride;
+  union pytorch_qnnp_conv_quantization_params quantization_params;
+  const pytorch_q8conv_ukernel_function ukernel;
+};
+
+static void compute_q8conv(
+    const struct q8conv_context context[1],
+    size_t group_index,
+    size_t image_index,
+    size_t mr_block_start,
+    size_t nr_block_start,
+    size_t group_range /* always 1 */,
+    size_t image_range /* always 1 */,
+    size_t mr_block_size,
+    size_t nr_block_size) {
+  const size_t bs = context->bs;
+  const size_t ks = context->ks;
+  const size_t kc = context->kc;
+  const size_t kc_stride = context->kc_stride;
+  const size_t m = context->m;
+  const size_t m_stride = context->m_stride;
+  const size_t n = context->n;
+  const size_t n_stride = context->n_stride;
+  const uint8_t** indirect_a = context->indirect_a;
+  const void* packed_w = context->packed_w;
+  uint8_t* c = context->c;
+  const size_t c_stride = context->c_stride;
+
+  context->ukernel(
+      mr_block_size,
+      nr_block_size,
+      kc,
+      ks,
+      indirect_a +
+          (mr_block_start + (image_index + group_index * bs) * m_stride) * ks,
+      (const void*)((uintptr_t)packed_w + (nr_block_start + group_index * n_stride) * (kc_stride * sizeof(uint8_t) + sizeof(int32_t))),
+      c + (mr_block_start + image_index * m) * c_stride + group_index * n +
+          nr_block_start,
+      c_stride,
+      &context->quantization_params);
+};
+
+struct QnnpackDeleter {
```

Comment:
can we move this struct to a common header now that's used in 2 places?

Reply:
I would love to. The problem is that the qnnpack library is compiled using C11, which doesn't support operator overloading. The `cc` files are compiled using CXX14, which does support it. That's why it is OK to keep the deleter here, but not in the common. I will look into it, because I agree that factoring it out is a good idea -- just need to find a way

Reply:
I will send a separate PR for this -- it looks like it is a little more involved.

Reply:
Please add a TODO or file an issue to track it

Reply:
This might sound like a rant, but let's prioritize deduplication of code. Even the current qnnpack state seems little unmaintainable.

Reply:
> Please add a TODO or file an issue to track it

Wil  send a separate PR -- we need to discuss what parts of qnnpack should be made available to PT

Reply:
Moved to the refactoring PR